0000-interledger-protocol-4/0000-interledger-protocol-4.md

```diff
@@ -0,0 +1,146 @@
+
+## Overview
+
+### Design Goals
+
+- **Neutrality** - Interledger is not tied to any company, currency, or network.
+- **Interoperability** - ILP should be usable across any type of ledger, even those that were not built for interoperability.
+- **Security** - Senders, connectors, and receivers should be protected from one another and especially isolated from risks posed by parties that they are indirectly connected to.
+- **Simplicity** - The core ILP should be as simple as possible. There is room for significant variability at lower and higher layers of the [Interledger stack](../0001-interledger-architecture/0001-interledger-architecture.md#interledger-protocol-suite), but the Interledger layer is the one part where widespread agreement is needed for interoperability.
+- **End-to-End Principle** - Inspired by the Internet, any features that do not need to be implemented by the core protocol and network should be built into the edges of the network (the sender and receiver).
+
+### Relation to Other Protocols
+
+ILPv4 packets may be sent over any communication channel between peers. Payment obligations created by ILP packets may be settled using any available means, ranging from ledger transfers triggered by an API, to signed updates to payment channels, or the physical delivery of cash or other goods.
+
+ILPv4 will probably be used with a Transport Layer protocol such as the Pre-Shared Key V2 (PSKv2) protocol, which handles the generation of ILP conditions and fulfillments.
+
+### Differences from Previous Versions of ILP
+
+- **Designed for Smaller, More Homogenous Packet Amounts** - ILPv4 applies a number of simplifications based on the change from supporting all packet amounts to focusing the Interledger layer on lower-value packets. Larger amounts can be sent using chunked payments, but the core network is optimized for sending large volumes of small packets. This renders unnecessary some of the more complex features of previous versions like Liquidity Curves for expressing how exchange rates may vary depending on the packet amount. Lower-value packets also help to minimize [connector risks](../0018-connector-risk-mitigations/0018-connector-risk-mitigations.md).
+- **Payment Channels, Not On-Ledger Escrow** - Since ILPv4 is optimized for smaller packets, speed and cost are of greater importantance than in previous versions. ILPv4 uses ledgers or payment channels for settling bilateral payment obligations but ILPv4 packets are sent just between connectors, rather than through the underlying ledgers themselves. This enables packet timeouts to be short, because they do not need to include the processing time of slower ledgers, which further reduces [connector risks](../0018-connector-risk-mitigations/0018-connector-risk-mitigations.md).
```

Comment:
in which sense is ILPv4 optimized for smaller packets? is the choice for unconditional payment channels a part of the ILPv4 design, or could someone do conditional payment channels and large packets, and still call it ILPv4?

Reply:
That's an interesting question. I think a lot of the things we've now built into or on top of ILPv4 assume the packets are relatively small, but you might still be able to do larger packets or conditional transfers.

Some of the assumptions are:
- Sending packets is free or of negligible cost, so we can use them liberally for things like quoting
- Ledger protocols are now mostly request/response, meaning the response isn't retried if it doesn't go through the first time
- Connectors don't store prepared packets in a db, because the assumption is that if your system dies you just lose all the packets that are in-flight
- Connectors have relatively low maximum hold times by default and as a connector you should be prepared to forward packets that have less than 30 seconds before they expire (you won't fare very well if you need to build in a long expiry window because your ledger is slower than that)
- As someone sending packets through the open Interledger, you won't be able to rely on the assumption that your payment can go through in one chunk and you might be told it's too big